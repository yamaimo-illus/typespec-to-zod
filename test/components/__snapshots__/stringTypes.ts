import { z } from "zod";

/*
 ██████╗ ██████╗ ███╗   ███╗██████╗  ██████╗ ███╗   ██╗███████╗███╗   ██╗████████╗███████╗
██╔════╝██╔═══██╗████╗ ████║██╔══██╗██╔═══██╗████╗  ██║██╔════╝████╗  ██║╚══██╔══╝██╔════╝
██║     ██║   ██║██╔████╔██║██████╔╝██║   ██║██╔██╗ ██║█████╗  ██╔██╗ ██║   ██║   ███████╗
██║     ██║   ██║██║╚██╔╝██║██╔═══╝ ██║   ██║██║╚██╗██║██╔══╝  ██║╚██╗██║   ██║   ╚════██║
╚██████╗╚██████╔╝██║ ╚═╝ ██║██║     ╚██████╔╝██║ ╚████║███████╗██║ ╚████║   ██║   ███████║
 ╚═════╝ ╚═════╝ ╚═╝     ╚═╝╚═╝      ╚═════╝ ╚═╝  ╚═══╝╚══════╝╚═╝  ╚═══╝   ╚═╝   ╚══════╝
*/
export const schemaStringTypes = z.object({
    standardProperty: z.string().default("default value"),
    urlProperty: z.string().url().default("https://example.com")
});
