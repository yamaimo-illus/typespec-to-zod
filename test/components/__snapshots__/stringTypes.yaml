openapi: 3.0.0
info:
  title: (title)
  version: 0.0.0
tags: []
paths: {}
components:
  schemas:
    StringTypes:
      type: object
      required:
        - standardProperty
        - urlProperty
      properties:
        standardProperty:
          type: string
          default: default value
        urlProperty:
          type: string
          format: uri
          default: https://example.com
